auth/internal/trustboundary/trust_boundary.go

```diff
@@ -0,0 +1,374 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustboundary
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+
+	"sync"
+
+	"cloud.google.com/go/auth"
+	"cloud.google.com/go/auth/internal"
+	"cloud.google.com/go/auth/internal/transport/headers"
+	"github.com/googleapis/gax-go/v2/internallog"
+)
+
+const (
+	// serviceAccountAllowedLocationsEndpoint is the URL for fetching allowed locations for a given service account email.
+	serviceAccountAllowedLocationsEndpoint = "https://iamcredentials.%s/v1/projects/-/serviceAccounts/%s/allowedLocations"
+)
+
+var (
+	// enabledOnce ensures the check for trust boundary is only performed once.
+	enabledOnce sync.Once
+	// enabled controls whether the trust boundary feature is enabled.
+	enabled    bool
+	enabledErr error
+)
+
+// IsEnabled returns if the trust boundary feature is enabled and an error if
+// the configuration is invalid.
+func IsEnabled() (bool, error) {
+	enabledOnce.Do(func() {
+		enabled, enabledErr = isTrustBoundaryEnabled()
+	})
+	return enabled, enabledErr
+}
```

Comment:
I believe you can replace the 15 lines above with a cleaner and safer implementation using https://pkg.go.dev/sync#OnceValues.

```go
// isEnabled wraps isTrustBoundaryEnabled with sync.OnceValues to ensure it's
// called only once.
var isEnabled = sync.OnceValues(isTrustBoundaryEnabled)

// IsEnabled returns if the trust boundary feature is enabled and an error if
// the configuration is invalid. The underlying check is performed only once.
func IsEnabled() (bool, error) {
	return isEnabled()
}
```

If you do this, I think you can then remove `TestIsEnabled` and `resetIsEnabled`, because there is essentially no logic for you to test.